SCEPman/Private/log-analytics.ps1

```diff
@@ -0,0 +1,470 @@
+    if($isAppServiceLinux) {
+        $WorkspaceIdVariable = "AppConfig__LoggingConfig__WorkspaceId"
+        $SharedKeyVariable = "AppConfig__LoggingConfig__SharedKey"
+    } else {
+        $WorkspaceIdVariable = "AppConfig:LoggingConfig:WorkspaceID"
+        $SharedKeyVariable = "AppConfig:LoggingConfig:SharedKey"
+    }
+    $null = Invoke-Az @("webapp", "config", "appsettings", "delete", "--name", $AppServiceName, "--resource-group", $ResourceGroup, "--setting-names", $WorkspaceIdVariable, $SharedKeyVariable)
+}
+
+function CreateLogAnalyticsWorkspace($ResourceGroup, $WorkspaceId) {
+    $workspaceAccount = GetLogAnalyticsWorkspace -ResourceGroup $ResourceGroup -WorkspaceId $WorkspaceId
+        if($null -eq $workspaceAccount) {
+            #Create a new workspace
+            Write-Information 'Log analytics workspace not found. We will create one now'
+            $workspaceName = $ResourceGroup.ToLower() -replace '[^a-z0-9]',''
+
+            # Length between 4-63, Alphanumerics and hyphens, Start and end with alphanumeric.
+            if($workspaceName.Length -gt 56) {
+                $workspaceName = $workspaceName.Substring(0,56)
+            }
+            $workspaceName = "log-$($workspaceName)-sc"
+            $potentialWorkspaceName = Read-Host "Please hit enter now if you want to create the log analytics workspace with name $workspaceName or enter the name of your choice, and then hit enter"
+            if($potentialWorkspaceName) {
+                $workspaceName = $potentialWorkspaceName
+            }
+            $workspaceAccount = Invoke-Az @("monitor", "log-analytics", "workspace", "create", "--resource-group", $ResourceGroup, "--name", $workspaceName, "--only-show-errors") | Convert-LinesToObject
+            if($null -eq $workspaceAccount) {
+                Write-Error 'Log analytics workspace not found and we are unable to create one. Please check logs for more details before re-running the script'
+                throw 'Log analytics workspace not found and we are unable to create one. Please check logs for more details before re-running the script'
+            }
+            Write-Information "Log analytics workspace $workspaceName created"
+            $workspaceAccount = GetLogAnalyticsWorkspace -ResourceGroup $ResourceGroup
+
+            if($null -eq $workspaceAccount) {
+                Write-Error 'Log analytics workspace not found after creation'
+                throw 'Log analytics workspace not found after creation'
+            }
+        }
+    return $workspaceAccount
+}
+
+function GetLogAnalyticsTable($ResourceGroup, $WorkspaceAccount, $SubscriptionId, $tableName) {
+    $table = Invoke-Az -MaxRetries 0 -azCommand @("monitor", "log-analytics", "workspace", "table", "show", "--resource-group", $ResourceGroup, "--workspace-name", $($WorkspaceAccount.name), "--name", $tableName) | Convert-LinesToObject
```

Comment:
Why the explicit MaxRetries 0? Is there some negative side effect when executing this multiple times or is it guaranteed to work on first try?